fastddsspy_participants/test/model/blackbox/ParticipantDatabaseTest.cpp

```diff
@@ -0,0 +1,401 @@
+    new_data->info.active = true;
+    ddspipe::core::types::Guid guid_data = ddspipe::core::testing::random_guid();
+    new_data->info.guid = guid_data;
+    dds_reader->simulate_data_reception(std::move(new_data));
+
+    // Wait
+    sleep(1);
+
+    // Check information
+    ASSERT_TRUE(model->participant_database_.at(guid_data).active);
+
+    ASSERT_EQ(model->participant_database_.size(), 1);
+}
+
+TEST(ParticipantDatabaseTest, active_false)
+{
+    // Create Model
+    std::shared_ptr<spy::participants::SpyModel> model =
+            std::make_shared<spy::participants::SpyModel>();
+
+    const ddspipe::core::types::ParticipantId id_ = "DDS Spy mock participant";
+
+    std::shared_ptr<test::SpyDdsParticipantMock> dds_participant =
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<ddspipe::core::DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->participant_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send message
+    auto new_data = std::make_unique<spy::participants::ParticipantInfoData>();
+    new_data->info.name = "participant::test";
+    new_data->info.active = false;
+    ddspipe::core::types::Guid guid_data = ddspipe::core::testing::random_guid();
+    new_data->info.guid = guid_data;
+    dds_reader->simulate_data_reception(std::move(new_data));
+
+    // Wait
+    sleep(1);
+
+    // Check information
+    ASSERT_FALSE(model->participant_database_.at(guid_data).active);
+
+    ASSERT_EQ(model->participant_database_.size(), 1);
+}
+
+TEST(ParticipantDatabaseTest, n_msgs_name)
+{
+    // Create Model
+    std::shared_ptr<spy::participants::SpyModel> model =
+            std::make_shared<spy::participants::SpyModel>();
+
+    const ddspipe::core::types::ParticipantId id_ = "DDS Spy mock participant";
+
+    std::shared_ptr<test::SpyDdsParticipantMock> dds_participant =
+            std::make_shared<test::SpyDdsParticipantMock>(id_);
+
+    // Create DDS Pipe
+    std::shared_ptr<ddspipe::core::DdsPipe> pipe = create_pipe(model, dds_participant);
+    pipe->enable();
+
+    auto dds_reader = dds_participant->participant_reader;
+    ASSERT_NE(dds_reader, nullptr);
+
+    // Send messages
+    for (unsigned int i = 0; i < 10; i++)
+    {
+        auto new_data = std::make_unique<spy::participants::ParticipantInfoData>();
+        new_data->info.name = "participant::test"+std::to_string(i);
+        new_data->info.active = true;
+        ddspipe::core::types::Guid guid_data = ddspipe::core::testing::random_guid();
+        new_data->info.guid = guid_data;
+        dds_reader->simulate_data_reception(std::move(new_data));
+    }
+
+    unsigned int i = 0;
+    for (const auto& participant : model->participant_database_)
+    {
+        ASSERT_EQ(participant.second.name, "participant::test"+std::to_string(i));
+        i++;
+    }
+
+    // Wait
+    sleep(1);
```

Comment:
I would rather use a variable, so if it changes in the future it must be only changed in one place. I would also suggest a much lower time, as this highly increments the time spent by the tests. 0.1 or 0.05 should be enough